piptools/scripts/sync.py

```diff
@@ -9,15 +10,21 @@
 from pip._internal.commands.install import InstallCommand
 from pip._internal.index.package_finder import PackageFinder
 from pip._internal.utils.misc import get_installed_distributions
+from pip._vendor.packaging.version import Version
 
 from .. import sync
 from .._compat import parse_requirements
 from ..exceptions import PipToolsError
 from ..logging import log
 from ..repositories import PyPIRepository
-from ..utils import flat_map
+from ..utils import (
+    flat_map,
+    get_pip_version_for_python_executable,
+    get_sys_path_for_python_executable,
+)
 
 DEFAULT_REQUIREMENTS_FILE = "requirements.txt"
+MIN_PIP_VERSION = Version("20.3")  # Should be in sync with version in setup.cfg
```

Comment:
Instead of hardcoding such things, it's possible to parse it from the metadata via `importlib_metadata` or `pkg_util` or smth like that.

Reply:
You may find some inspiration here https://github.com/cherrypy/cherrypy/blob/e00d162/docs/conf.py#L32-L57

Reply:
thanks for the suggestion, updated to use `get_distribution` and compare the version with the specifier